elastic/beats
# Fix clean_inactive for Filestream input

## Proposed commit message

The `clean_inactive` parameter was being parsed with the wrong key. It
is parsed/used by an anonymous struct on `input-logfile/manager.go`,
there it was parsed and used as `CleanTimeout` (`clean_timeout`). This
`CleanTimeout` setting has got exactly the same effect as the
`clean_inactive` described in our documentation.

This commit fixes this bug by renaming `clean_timeout` to
`clean_inactive` so the configuration value can have effect.

## Checklist


## Author's Checklist

- [x] Wait for https://github.com/elastic/beats/pull/38488 to be merged and rebase onto `main`

## How to test this PR locally
All files are at the end of this section.
1. Start Filebeat using `filebeat.yml`
2. Create the file to be ingested/write some data
    ```
    echo "foo bar" > /tmp/log.log
    ```
3. Look at Filebeat logs to ensure the file has been found, ingested and the reader was closed. Look for logs messages like those:
    ```
    A new file /tmp/log.log has been found
    Reader was closed. Closing. Path='/tmp/log.log
    ```
4. Ensure the store clean up run and removed the only entry, look for a log message like:
    ```
    1 entries removed
    ```
5. Ensure the registry log file (`data/registry/filebeat/log.json`) contains an remove operation like this:
    ```
    {"op":"remove","id":5}
    ```

<details><summary>filebeat.yml</summary>
<p>

```yaml
filebeat.inputs:
  - type: filestream
    id: "test-clean-inactive"
    paths:
      - /tmp/log.log

    clean_inactive: 3s
    ignore_older: 2s
    close.on_state_change.inactive: 1s
    prospector.scanner.check_interval: 1s

filebeat.registry:
  cleanup_interval: 5s
  flush: 1s

queue.mem:
  events: 32
  flush.min_events: 8
  flush.timeout: 0.1s

path.home: %s

output.file:
  path: ${path.home}
  filename: "output-file"
  rotate_every_kb: 10000

logging:
  level: debug
  selectors:
    - input
    - input.filestream
  metrics:
    enabled: false
```

</p>
</details> 


## Related issues
- Closes: https://github.com/elastic/beats/issues/36629
- Needs: https://github.com/elastic/beats/pull/38488

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~
